Sources/UIComponent/Components/StackComponent.swift

```diff
@@ -99,26 +162,101 @@ extension StackComponent {
+                    return (point(main: offset + main(frame.point), cross: cross(frame.point)), frame.size)
+                }
+            } else if justifyContent == .spaceBetween {
+                let leftOverPrimary = main(constraint.maxSize) - maxTotalRowMainSize
+                guard frames.count > 1 else { break }
+                let spacing = leftOverPrimary / CGFloat(frames.count - 1)
+                adjustFrames = adjustFrames.enumerated().map { (index, frame) -> (CGPoint, CGSize) in
+                    // clear spacing
+                    var main = main(frame.point) - (distributedSpacing * CGFloat(index))
+                    // added new spacing
+                    main += CGFloat(index) * spacing
+                    return (point(main: main, cross: cross(frame.point)), frame.size)
+                }
+            } else if justifyContent == .spaceEvenly {
+                let leftOverPrimary = main(constraint.maxSize) - maxTotalRowMainSize
+                let spacing = leftOverPrimary / CGFloat(frames.count+1)
+                adjustFrames = adjustFrames.enumerated().map { (index, frame) -> (CGPoint, CGSize) in
+                    // clear spacing
+                    var main = main(frame.point) - (distributedSpacing * CGFloat(index))
+                    // added new spacing
+                    main += CGFloat(index+1) * spacing
+                    return (point(main: main, cross: cross(frame.point)), frame.size)
+                }
+            } else if justifyContent == .spaceAround {
+                let leftOverPrimary = main(constraint.maxSize) - maxTotalRowMainSize
+                let spacing = leftOverPrimary / CGFloat(frames.count)
+                adjustFrames = adjustFrames.enumerated().map { (index, frame) -> (CGPoint, CGSize) in
+                    // clear spacing
+                    var main = main(frame.point) - (distributedSpacing * CGFloat(index))
+                    // added new spacing
+                    if index == 0 {
+                        main += spacing/2
+                    } else {
+                        main += (CGFloat(index) * spacing) + (spacing/2)
+                    }
+                    return (point(main: main, cross: cross(frame.point)), frame.size)
+                }
+            }
+            if alignItems == .end {
+                adjustFrames = adjustFrames.map { frame -> (CGPoint, CGSize) in
+                    return (point(main: main(frame.point), cross: (maxCurrenRowCrossSize - cross(frame.size)) + cross(frame.point)), frame.size)
+                }
+            } else if alignItems == .center {
+                adjustFrames = adjustFrames.map { frame -> (CGPoint, CGSize) in
+                    return (point(main: main(frame.point), cross: ((maxCurrenRowCrossSize - cross(frame.size)) / 2) + cross(frame.point)), frame.size)
+                }
+            }
+            rowFrames[row] = adjustFrames
+        }
```

Comment:
This seems a little complicated. I think we can merge the HStack with the Flow layout somehow and merge the best from both world. Can you take a look at how the `Flow` layout distributes its row items using alignItems?

Reply:
Flow layout to wait awhile now trying to do `.flex` modifier